src/System.Text.Json/src/System/Text/Json/Node/JsonBoolean.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    ///   Represents a boolean JSON value.
+    /// </summary>
+    public class JsonBoolean : JsonNode, IEquatable<JsonBoolean>
+    {
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonBoolean"/> class representing the value <see langword="false"/>.
+        /// </summary>
+        public JsonBoolean() => Value = false;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonBoolean"/> class representing a specified value.
+        /// </summary>
+        public JsonBoolean(bool value) => Value = value;
+
+        /// <summary>
+        ///   Gets or sets the boolean value represented by the instance.
+        /// </summary>
+        public bool Value { get; set; }
+
+        /// <summary>
+        ///   Converts a <see cref="bool"/> to a JSON boolean.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        public static implicit operator JsonBoolean(bool value) => new JsonBoolean(value);
```

Comment:
What are the use cases for this?  There's an implicit allocation that happens here; just want to make sure we've thought it through.

Reply:
We are still discussing it, I created an issue here: https://github.com/dotnet/corefx/issues/40118

Reply:
@stephentoub the main idea of having these implicit operators where so that we could enable great experience for initializing a JsonObject using Collection initialization such that it looked as close as possible to the actual JSON, without the need to use the `new` keyword everywhere.